rocAL/include/decoders/image/rocjpeg_decoder.h

```diff
@@ -0,0 +1,139 @@
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+#include "decoders/image/decoder.h"
+#include "rocjpeg.h"
+#include "decoders/image/rocjpeg_decoder.h"
+
+struct ScalingFactor {
+  unsigned num;
+  unsigned denom;
+};
+
+class HWRocJpegDecoder : public Decoder {
+   public:
+    //! Default constructor
+    HWRocJpegDecoder();
+#if ENABLE_HIP
+    HWRocJpegDecoder(hipStream_t &stream) { _hip_stream = stream; }
+#endif
+    //! Decodes the header of the Jpeg compressed data and returns basic info about the compressed image
+    /*!
+     \param input_buffer  User provided buffer containig the encoded image
+     \param input_size Size of the compressed data provided in the input_buffer
+     \param width pointer to the user's buffer to write the width of the compressed image to
+     \param height pointer to the user's buffer to write the height of the compressed image to
+     \param color_comps pointer to the user's buffer to write the number of color components of the compressed image to
+    */
+    Status decode_info(unsigned char *input_buffer, size_t input_size, int *width, int *height, int *color_comps) override;
+    
+    //! Decodes the header of the Jpeg compressed data and returns basic info about the compressed image
+    //! It also scales the width and height wrt max decoded width and height
+    /*!
+     \param input_buffer  User provided buffer containig the encoded image
+     \param input_size Size of the compressed data provided in the input_buffer
+     \param width pointer to the user's buffer to write the width of the compressed image/scaled width based on max_decoded_width
+     \param height pointer to the user's buffer to write the height of the compressed image/scaled height based on max_decoded_height
+     \param actual_width pointer to the user's buffer to write the width of the compressed image
+     \param actual_height pointer to the user's buffer to write the height of the compressed image
+     \param max_decoded_width maximum width of the decoded image
+     \param max_decoded_height maximum height of the decoded image
+     \param desired_decoded_color_format user provided color format of the decoded image
+     \param index index of the image in the batch for which decode info must be fetched
+    */
+    Status decode_info(unsigned char *input_buffer, size_t input_size, int *width, int *height, int *actual_width, int *actual_height, int max_decoded_width, int max_decoded_height, Decoder::ColorFormat desired_decoded_color_format, int index) override;
+
+    Decoder::Status decode(unsigned char *input_buffer, size_t input_size, unsigned char *output_buffer,
```

Comment:
Mention in all instances if any of the input_buffer is required as a HIP buffer. Mention like
"User provided HIP buffer ..."